arch/arm/soc/atmel_sam/sam4s/soc.h

```diff
@@ -13,207 +13,26 @@
 #ifndef _ATMEL_SAM4S_SOC_H_
 #define _ATMEL_SAM4S_SOC_H_
 
-/* IRQ numbers (from section 11.1, Peripheral Identifiers). */
-#define IRQ_SUPC	0	/* Supply Controller                    */
-#define IRQ_RSTC	1	/* Reset Controller                     */
-#define IRQ_RTC		2	/* Real-time Clock                      */
-#define IRQ_RTT		3	/* Real-time Timer                      */
-#define IRQ_WDG		4	/* Watchdog Timer                       */
-#define IRQ_PMC		5	/* Power Management Controller          */
-#define IRQ_EEFC0	6	/* Enhanced Embedded Flash Controller 0 */
-#define IRQ_EEFC1	7	/* Enhanced Embedded Flash Controller 1 */
-#define IRQ_UART0	8	/* UART                                 */
-#define IRQ_UART1	9	/* UART                                 */
-#define IRQ_SMC		10	/*                                  */
-#define IRQ_PIOA	11	/* Parallel IO Controller A             */
-#define IRQ_PIOB	12	/* Parallel IO Controller B             */
-#define IRQ_PIOC	13	/* Parallel IO Controller C             */
-#define IRQ_USART0	14	/* USART #0                             */
-#define IRQ_USART1	15	/* USART #1                             */
-#define IRQ_HSMCI	18	/* High Speed Multimedia Card Interface */
-#define IRQ_TWI0	19	/* Two-wire Interface #0                */
-#define IRQ_TWI1	20	/* Two-wire Interface #1                */
-#define IRQ_SPI		21	/* SPI                                  */
-#define IRQ_SSC		22	/* Synchronous Serial Controller        */
-#define IRQ_TC0		23	/* Timer Counter Channel #0             */
-#define IRQ_TC1		24	/* Timer Counter Channel #1             */
-#define IRQ_TC2		25	/* Timer Counter Channel #2             */
-#define IRQ_TC3		26	/* Timer Counter Channel #3             */
-#define IRQ_TC4		27	/* Timer Counter Channel #4             */
-#define IRQ_TC5		28	/* Timer Counter Channel #5             */
-#define IRQ_ADC		29	/* ADC Controller                       */
-#define IRQ_DACC	30	/* DAC Controller                       */
-#define IRQ_PWM		31	/* PWM Controller                       */
-#define IRQ_CRCCU	32	/* CRC Controller                       */
-#define IRQ_ACC		33	/*                    */
-#define IRQ_UDP		34	/* USB Device Port                      */
+#define DONT_USE_CMSIS_INIT
```

Comment:
This is probably going to need:

#ifndef _ASMLANGUAGE

...
#endif 

The SAME7 code does this in board.h which is wrong and should be cleaned up.